istioctl/cmd/checkinject.go

```diff
@@ -0,0 +1,282 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	admit_v1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"istio.io/api/label"
+	"istio.io/istio/istioctl/pkg/util/handlers"
+	"istio.io/istio/istioctl/pkg/writer/table"
+	analyzer_util "istio.io/istio/pkg/config/analysis/analyzers/util"
+)
+
+var labelPairs string
+
+func checkInjectCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "check-inject [<type>/]<name>[.<namespace>]",
+		Short: "Check the injection status or inject-ability of a given resource, explains why it is (or will be) injected or not",
+		Long: `
+Checks associated resources of the given resource, and running webhooks to examine whether the pod can be or will be injected or not.`,
+		Example: `	# Check the injection status of a pod
+	istioctl check-inject details-v1-fcff6c49c-kqnfk.test
```

Comment:
for newly added command, it is better to add them under `experimental` at first

Reply:
Thanks, fixed.